examples/transition.rs

```diff
@@ -0,0 +1,192 @@
+        .args(["-protocol_whitelist", "file,rtp,udp", &output_sdp])
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()?;
+
+    info!("[example] Send register output request.");
+    common::post(&json!({
+        "type": "register",
+        "entity_type": "output_stream",
+        "output_id": "output_1",
+        "port": 8002,
+        "ip": "127.0.0.1",
+        "resolution": {
+            "width": VIDEO_RESOLUTION.width,
+            "height": VIDEO_RESOLUTION.height,
+        },
+        "encoder_settings": {
+            "preset": "ultrafast"
+        }
+    }))?;
+
+    info!("[example] Send register input request.");
+    common::post(&json!({
+        "type": "register",
+        "entity_type": "input_stream",
+        "input_id": "input_1",
+        "port": 8004
+    }))?;
+
+    let shader_source = include_str!("../compositor_render/examples/silly/silly.wgsl");
+    info!("[example] Register shader transform");
+    common::post(&json!({
+        "type": "register",
+        "entity_type": "shader",
+        "shader_id": "example_shader",
+        "source": shader_source,
+        "fallback_strategy": "fallback_if_all_inputs_missing",
+        "constraints": [
+            {
+                "type": "input_count",
+                "fixed_count": 1,
+            }
+        ]
+    }))?;
+
+    info!("[example] Register static image");
+    common::post(&json!({
+        "type": "register",
+        "entity_type": "image",
+        "image_id": "example_image",
+        "asset_type": "gif",
+        "url": "https://gifdb.com/images/high/rust-logo-on-fire-o41c0v9om8drr8dv.gif",
+    }))?;
+
+    info!("[example] Start pipeline");
+    common::post(&json!({
+        "type": "start",
+    }))?;
+
+    info!("[example] Start input stream");
+    let ffmpeg_source = format!(
+        "testsrc=s={}x{}:r=30,format=yuv420p",
+        VIDEO_RESOLUTION.width, VIDEO_RESOLUTION.height
+    );
+    Command::new("ffmpeg")
+        .args([
+            "-re",
+            "-f",
+            "lavfi",
+            "-i",
+            &ffmpeg_source,
+            "-c:v",
+            "libx264",
+            "-f",
+            "rtp",
+            "rtp://127.0.0.1:8004?rtcpport=8004",
+        ])
+        .spawn()?;
+
+    info!("[example] Update scene");
+    common::post(&json!({
+        "type": "update_scene",
+        "nodes": [
+            {
+                "node_id": "image",
+                "type": "image",
+                "image_id": "example_image",
+            },
+            {
+                "type": "shader",
+                "node_id": "shader_1",
+                "shader_id": "example_shader",
+                "fallback_id": "image",
+                "input_pads": [
+                    "input_1",
+                ],
+                "resolution": { "width": 960, "height": 540 },
+            },
+            {
+                "node_id": "transition_1",
+                "type": "transition",
```

Comment:
> I know that JSON objects are unordered

Technically keys in json object are ordered :D

Reply:
Any resource for that? 
RFCs says otherwise
https://datatracker.ietf.org/doc/html/rfc8259,  https://www.rfc-editor.org/rfc/rfc7159.html
```
   An object is an unordered collection of zero or more name/value
   pairs, where a name is a string and a value is a string, number,
   boolean, null, object, or array.
```

Reply:
JSON (data format) - Depends on your definition. It's text-based, so you can see the order of the fields so it's ordered. On the other hand, JSONs are considered equal if their field order does not match, so in this sense keys are unordered.

Javascript object - I'm not sure if the standard guarantees it, but in every implementation, I'm aware that JS guarantees the order of keys in the object e.g. `Object.keys()` gives a deterministic result.